tools/shared.py

```diff
@@ -576,25 +581,21 @@ def check_sanity(force=False):
 
     with ToolchainProfiler.profile_block('sanity compiler_engine'):
       if not jsrun.check_engine(COMPILER_ENGINE):
-        logging.critical('The JavaScript shell used for compiling (%s) does not seem to work, check the paths in %s' % (COMPILER_ENGINE, EM_CONFIG))
-        sys.exit(1)
+        exit_with_error('The JavaScript shell used for compiling (%s) does not seem to work, check the paths in %s', COMPILER_ENGINE, EM_CONFIG)
 
     with ToolchainProfiler.profile_block('sanity LLVM'):
       for cmd in [CLANG, LLVM_LINK, LLVM_AR, LLVM_OPT, LLVM_AS, LLVM_DIS, LLVM_NM, LLVM_INTERPRETER]:
         if not os.path.exists(cmd) and not os.path.exists(cmd + '.exe'):  # .exe extension required for Windows
-          logging.critical('Cannot find %s, check the paths in %s' % (cmd, EM_CONFIG))
-          sys.exit(1)
+          exit_with_error('Cannot find %s, check the paths in %s', cmd, EM_CONFIG)
 
     if not os.path.exists(PYTHON) and not os.path.exists(cmd + '.exe'):
       try:
-        run_process([PYTHON, '--version'], stdout=PIPE, stderr=PIPE)
+        run_process([PYTHON, '--xversion'], stdout=PIPE, stderr=PIPE)
       except:
-        logging.critical('Cannot find %s, check the paths in %s' % (PYTHON, EM_CONFIG))
-        sys.exit(1)
+        exit_with_error('Cannot find %s, check the paths in %s', PYTHON, EM_CONFIG)
```

Comment:
We're losing a slight bit of information here, by changing these criticals to errors
Maybe a boolean kwarg, `is_critical=True`?